src/lib/badge/badge.ts

```diff
@@ -248,4 +251,17 @@ export class MatBadge extends _MatBadgeMixinBase implements OnDestroy, OnChanges
       }
     }
   }
+
+  /** Creates string according to max value if one is set */
+  private _applyMax(): string {
+    let numericContent: number = Number(this.content);
```

Comment:
This assumes that the content will be a number, however we allow any kind of string. E.g. in the dev app there are some examples with emoji.

Reply:
I am trying to check if the `content` is `Number`. I am not able to use `isNaN` in typescript as it accepts only `Number`. This is why I am calling:
`let numericContent: number = Number(this.content);`

`Number(input)` will return number if `input` is number or empty string and `NaN` in all other cases. Then I am calling `isNaN` over the `numericContent`. Hope this makes sense :)